clients/go/zbc/credentialsProvider.go

```diff
@@ -30,75 +33,122 @@ type CredentialsProvider interface {
-//	endpoint:
-//		auth:
-//			credentials:
-//				access_token: <token>,
-//				token_type: <prefix>,
-type ZeebeClientCredentialsProvider struct {
-	zeebeCredentialsPath string
+// A built-in CredentialsProvider that contains contains credentials obtained from an OAuth authorization server,
+// including a token prefix and an access token. Using these values it sets the 'Authorization' header of each gRPC call.
+type OAuthCredentialsProvider struct {
+	Credentials *OauthCredentials
 }
 
-type zeebeClientCredentials struct {
-	Endpoint struct {
-		Auth struct {
-			Credentials struct {
-				AccessToken string `yaml:"access_token"`
-				ExpiresIn   string `yaml:"expires_in"`
-				TokenType   string `yaml:"token_type"`
-			}
-		}
-	}
+// Configuration data for the OAuthCredentialsProvider, containing the required data to request an access token from
+// an OAuth authorization server which will be appended to each gRPC call's headers.
+type OAuthProviderConfig struct {
+	// The client identifier used to request an access token. Can be overridden with the environment variable 'ZEEBE_CLIENT_ID'.
+	ClientId string
+	// The client secret used to request an access token. Can be overridden with the environment variable 'ZEEBE_CLIENT_SECRET'.
+	ClientSecret string
+	// The audience to which the access token will be sent. Can be overridden with the environment variable 'ZEEBE_TOKEN_AUDIENCE'.
+	Audience string
+	// The URL for the authorization server from which the access token will be requested. Can be overridden with
+	// the environment variable 'ZEEBE_AUTHORIZATION_SERVER_URL'.
+	AuthorizationServerUrl string
+}
+
+type OauthCredentials struct {
+	AccessToken string `json:"access_token"`
+	ExpiresIn   uint64 `json:"expires_in"`
+	TokenType   string `json:"token_type"`
+	Scope       string `json:"scope"`
+}
+
+type oauthRequestPayload struct {
+	ClientId     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+	Audience     string `json:"audience"`
+	GrantType    string `json:"grant_type"`
 }
 
 // Takes a map of headers as input and adds an access token prefixed by a token type to the 'Authorization'
-// header of a gRPC call. The access token and the token type are obtained from a Zeebe credentials YAML file that
-// should be found at the provided zeebeCredentialsPath.
-//
-// To use this with a JSON Web Token (JWT), the Zeebe credentials file would look like:
-//	endpoint:
-//		auth:
-//			credentials:
-//				access_token: jjjjj.wwwww.ttttt,
-//				token_type: Bearer,
-func (provider *ZeebeClientCredentialsProvider) ApplyCredentials(headers map[string]string) {
-	file, err := os.Open(provider.zeebeCredentialsPath)
-	if err != nil {
-		log.Fatalf("Failed to open the zeebe credentials file: %s", err)
-	}
+// header of a gRPC call.
+func (provider *OAuthCredentialsProvider) ApplyCredentials(headers map[string]string) {
+	tokenType := strings.TrimSpace(provider.Credentials.TokenType)
```

Comment:
I get the idea, but I would really just pass what the user is giving us; trimming space is a more a problem of whoever is constructing this struct, imo (e.g. `zbctl`), not of the client itself.